src/common/util/feature-flags.js

```diff
@@ -5,45 +5,28 @@
 import { ee } from '../event-emitter/contextual-ee'
 import { dispatchGlobalEvent } from '../dispatch/global-event'
 
-const expectedFlags = ['stn', 'err', 'ins', 'spa', 'sr']
-const noFlagFeatures = ['xhr', 'pvt']
-
 const sentIds = new Set()
 
-/** Note that this function only processes each unique flag ONCE, with the first occurrence of each flag and numeric value determining its switch on/off setting. */
+/**
+ * Sets the activatedFeatures object, dispatches the global loaded event,
+ * and emits the rumresp flag to features
+ * @param {{[key:string]:number}} flags key-val pair of flag names and numeric
+ * @param {string} agentIdentifier agent instance identifier
+ * @returns {void}
+ */
 export function activateFeatures (flags, agentIdentifier) {
   const sharedEE = ee.get(agentIdentifier)
   activatedFeatures[agentIdentifier] ??= {}
   if (!(flags && typeof flags === 'object')) return
   if (sentIds.has(agentIdentifier)) return
 
-  /** Flags returned from RUM call */
-  Object.entries(flags).forEach(([flag, num]) => {
-    emitFlag(flag, num)
-  })
-
-  /** We gate all features behind a flag for consistency, but some features dont require entitlements before running
-   * So we make our own flags and return them as truthy for now
-  */
-  noFlagFeatures.forEach(flag => {
-    emitFlag(flag, 1)
-  })
-
-  /** Any features that happen to fail to get a flag back from the RUM call would be blocking the rest of the features from operating, so return 0 by default */
-  expectedFlags.forEach(flag => {
-    emitFlag(flag, 0)
-  })
+  sharedEE.emit('rumresp', [flags])
+  activatedFeatures[agentIdentifier] = flags
```

Comment:
What do you think about throwing an `Object.freeze()` on `flags` for good measures? 😆 🤔

Reply:
Why?

Reply:
All referring to the same object. Presumably it could get mutated somewhere.

Reply:
I went ahead and added it since this is a public object and affects feature behavior

Reply:
nvm, freeze breaks ie11.  removing for now and we can revisit again another time